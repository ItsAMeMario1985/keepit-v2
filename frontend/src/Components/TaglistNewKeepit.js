import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { CloudIcon, UserIcon, TagMinusIcon, TagPlusIcon } from './Icons'
import LoadingSpinner from './LoadingSpinner'

export default function TaglistNewKeepit({
  onClick,
  tags,
  showIsCustom,
  showIsloading,
}) {
  function IsCustomIcon(tag) {
    if (showIsCustom) {
      if (tag.isCustom) {
        return <UserIcon />
      } else {
        return <CloudIcon fill="#fff" />
      }
    } else {
      return ''
    }
  }
  function IsAdded(tag) {
    if (!tag.added) {
      return <TagPlusIcon />
    } else {
      return <TagMinusIcon />
    }
  }

  if (tags.length === 0 && showIsloading) {
    return <LoadingSpinner />
  }

  return (
    <StyledTagList>
      {tags &&
        tags.map((tag, index) => (
          <StyledLi
            bgcolor={
              tag.added ? 'var(--color-primary)' : 'var(--color-tertiary)'
            }
            key={tag.value}
            onClick={() => onClick(tag.value, !tag.added, tag.isCustom)}
          >
            {IsAdded(tag)}
            <StyledTagText>{tag.value}</StyledTagText>
            <IsCustomIcon isCustom={tag.isCustom}></IsCustomIcon>
          </StyledLi>
        ))}
    </StyledTagList>
  )
}

const StyledTagText = styled.span`
  font-weight: 400;
  font-size: 14px;
  letter-spacing: 0.5px;
  margin: 0 8px 0 5px;
`

const StyledTagList = styled.ul`
  list-style-type: none;
  color: white;
  display: flex;
  flex-wrap: wrap;
  gap: 5px;
  min-height: 24px;
`

const StyledLi = styled.li`
  background-color: ${(props) => props.bgcolor};
  transition: background-color 0.4s ease-out;
  display: inline-block;
  border-bottom-right-radius: 8px;
  border-top-right-radius: 8px;
  min-width: 50px;
  display: inline-flex;
  align-items: center;
  padding: 5px 10px 5px 5px;
  border-left: 2px solid #e3e3e3;
  cursor: pointer;
  font-weight: 400;
  font-size: 14px;
`

const StyledLoading = styled.div``
