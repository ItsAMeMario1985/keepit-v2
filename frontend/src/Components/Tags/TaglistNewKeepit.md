```jsx
<TaglistNewKeepit
  tags={[{ value: 'TestTag', added: false, isCustom: false }]}
></TaglistNewKeepit>
```
