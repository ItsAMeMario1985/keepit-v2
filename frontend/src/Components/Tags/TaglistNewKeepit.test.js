import { render } from '@testing-library/react'
import TaglistNewKeepit from './TaglistNewKeepit'
describe('TaglistNewKeepit', () => {
  const tags = [
    {
      value: 'Beer',
      added: false,
      isCustom: false,
    },
    {
      value: 'Barware',
      added: false,
      isCustom: false,
    },
  ]
  it('renders correctly', () => {
    const { container } = render(
      <TaglistNewKeepit
        tags={tags}
        bgColor="var(--color-primary)"
        showIsCustom={true}
        showIsloading={true}
      ></TaglistNewKeepit>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
