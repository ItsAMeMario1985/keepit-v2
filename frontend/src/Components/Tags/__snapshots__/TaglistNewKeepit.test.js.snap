// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaglistNewKeepit renders correctly 1`] = `
<ul
  class="TaglistNewKeepit__StyledTagList-sc-1jvl86a-2 cfNsvs"
>
  <li
    class="TaglistNewKeepit__StyledLi-sc-1jvl86a-3 jJgOsw"
  >
    <svg>
      tag-plus.svg
    </svg>
    <span
      class="TaglistNewKeepit__StyledTagText-sc-1jvl86a-1 dIgmxD"
    >
      Beer
    </span>
    <svg
      fill="#fff"
    >
      cloud.svg
    </svg>
  </li>
  <li
    class="TaglistNewKeepit__StyledLi-sc-1jvl86a-3 jJgOsw"
  >
    <svg>
      tag-plus.svg
    </svg>
    <span
      class="TaglistNewKeepit__StyledTagText-sc-1jvl86a-1 dIgmxD"
    >
      Barware
    </span>
    <svg
      fill="#fff"
    >
      cloud.svg
    </svg>
  </li>
</ul>
`;
